src/PrestaShopBundle/Controller/Admin/FrameworkBundleAdminController.php

```diff
@@ -544,4 +561,10 @@ protected function getErrorMessageForException(Exception $e, array $messages)
             $e->getMessage()
         );
     }
+
+    #[Required]
+    public function setTranslator(TranslatorInterface $translator)
```

Comment:
Preper addin a protected `getTranslator` method that use `$this->container->get(TranslatorInterface)` This will avoid having to change all the controllers in module to force them being services via autowiring thus reducing the breaking change
Or if you don't want to add a new method you can use this direct in method trans and getFormErrorsForJS